bt_ddos_shield/state_manager.py

```diff
@@ -0,0 +1,83 @@
+from abc import ABC, abstractmethod
+from enum import Enum
+
+from bt_ddos_shield.domain import Domain
+
+
+class State:
+    """
+    Class representing state of MinerShield.
+    """
+
+    class ShieldPhase(Enum):
+        """
+        Possible phases of shield.
+        """
+        DISABLED = "disabled"                         # disabled - initial state
+        MANIFEST_PUBLISHED = "manifest_published"     # manifest is saved to storage
+        MANIFEST_BROADCASTED = "manifest_broadcasted" # info about manifest is published to blockchain
+        ENABLED = "enabled"                           # shield is enabled
+
+    phase: ShieldPhase                # current phase of the shield
+    banned_validators: dict[str, int] # banned validators with ban time (HotKey -> ban timestamp)
+    active_domains: dict[str, Domain] # active domains (validator HotKey -> Domain created for him)
+
+    def __init__(self):
+        self.phase = self.ShieldPhase.DISABLED
+        self.banned_validators = {}
+        self.active_domains = {}
+
+class StateManager(ABC):
+    """
+    Abstract base class for manager handling state of MinerShield.
+    """
```

Comment:
```suggestion
class AbstractMinerShieldStateManager(ABC):
```

Reply:
If I should add Abstract before all other abstract classes names?

Reply:
That's what I always do, makes things very clear. Clarity is good.